# HackBio-Genomics-Workshop-2022
The duration of this workshop spanned 5 weeks and was split into 5 stages. . All tasks and projects executed in the workshop are in this repository.
# Stage 0(Week 1) 
I was introduced to linux and performed fundamental tasks in BASh. 
The first task was to assign my first name and surname to a variable and print my full name.
Also, I further executed basic commands like creating and changing directory and completed 2 BASh stories.
 You can find my codes in Stage_0.sh

# Stage 1 (Week 2)
I learnt how to process biological data in a conda environment using fastqc, gasp and multiqc. My codes can be found in stage1.sh

# Stage2 (Week 3)
This week's task was focused on replicating oncogenomics tutorials in linux and Galaxy. I processed normal and tumor datasets and performed variants calling and annotation.
My codes are in Stage2_task.sh. 
Also, there was a live coding assessment session where I used bcftools to merge files. see livecodingtest.sh

# Stage 3 (Week 4&5)
PROJECT TOPIC ; PHARMACOGENOMICS OF HUMAN LEUKOCYTES ANTIGEN (HLA) VARIANTS IN 4 ASIAN GROUPS

I chose a project in precision medicine. The task was to compare HLA variations between 4 Asian population groups (Dai, Han, Southern Han Chinese, and Vietnamese) in the 1000 genomes database by performing Principal Components Analysis (PCA), Multidimensional Scaling (MDS) Analysis, and Linkage Disequilibrium (LD). However, I could not perform LD as the dataset was lacking phenotypic information.

 
Technical Information

PCA and MDS were carried out in Linux using PLINK. PCA and MDS plots were generated in RStudio (see plots in Rplot files above). For further information,

see Technical Specification _project3 

see pharmocogenomics_project.sh for Linux codes

See pharmacogenomics-PCA&MDS.R for R codes

Reference Tutorials

https://www-users.york.ac.uk/~dj757/popgenomics/workshop6.html

http://hpc.ilri.cgiar.org/beca/training/data_mgt_2017/BackgroundMaterial/PlinkTutorial.pdf
